src/tests/conduit/t_conduit_execution.cpp

```diff
@@ -0,0 +1,320 @@
+//     conduit::execution::forall<reduce_policy>(0, size, [=](index_t i)
+//     {
+//         sum_reducer += dev_vals_ptr[i];
+//     });
+//     // CONDUIT_DEVICE_ERROR_CHECK();
+
+//     EXPECT_EQ(sum_reducer.get(),5);
+
+
+//     // // min
+//     // ascent::ReduceMin<reduce_policy,index_t> min_reducer;
+//     // conduit::execution::forall<reduce_policy>(0, size, [=](index_t i)
+//     // {
+//     //     min_reducer.min(dev_vals_ptr[i]);
+//     // });
+//     // // CONDUIT_DEVICE_ERROR_CHECK();
+
+//     // EXPECT_EQ(min_reducer.get(),-10);
+
+//     // // minloc
+//     // ascent::ReduceMinLoc<reduce_policy,index_t> minloc_reducer;
+//     // conduit::execution::forall<reduce_policy>(0, size, [=](index_t i)
+//     // {
+//     //     minloc_reducer.minloc(dev_vals_ptr[i],i);
+//     // });
+//     // // CONDUIT_DEVICE_ERROR_CHECK();
+
+//     // EXPECT_EQ(minloc_reducer.get(),-10);
+//     // EXPECT_EQ(minloc_reducer.getLoc(),1);
+
+
+//     // // max
+//     // ascent::ReduceMax<reduce_policy,index_t> max_reducer;
+//     // conduit::execution::forall<reduce_policy>(0, size, [=](index_t i)
+//     // {
+//     //     max_reducer.max(dev_vals_ptr[i]);
+//     // });
+//     // // CONDUIT_DEVICE_ERROR_CHECK();
+
+//     // EXPECT_EQ(max_reducer.get(),10);
+
+//     // // maxloc
+//     // ascent::ReduceMaxLoc<reduce_policy,index_t> maxloc_reducer;
+//     // conduit::execution::forall<reduce_policy>(0, size, [=](index_t i)
+//     // {
+//     //     maxloc_reducer.maxloc(dev_vals_ptr[i],i);
+//     // });
+//     // // CONDUIT_DEVICE_ERROR_CHECK();
+
+//     // EXPECT_EQ(maxloc_reducer.get(),10);
+//     // EXPECT_EQ(maxloc_reducer.getLoc(),2);
+
+//     device_free(dev_vals_ptr);
+// }
+
+//-----------------------------------------------------------------------------
+// TEST(conduit_execution, cpp_magic_tests)
+TEST(conduit_execution, justin_fun)
+{
+    std::cout << "forall cases!" << std::endl;
+
+    conduit::execution::ExecutionPolicy serial = conduit::execution::ExecutionPolicy::serial();
+#if defined(CONDUIT_USE_OPENMP)
+    conduit::execution::ExecutionPolicy openmp = conduit::execution::ExecutionPolicy::openmp();
+#endif
+#if defined(CONDUIT_USE_RAJA)
+    conduit::execution::ExecutionPolicy device = conduit::execution::ExecutionPolicy::device();
+#endif
+
+    int size = 4;
+
+    conduit::execution::forall(serial, 0, size, [=] (int i)
+    {
+       std::cout << i << std::endl;
+    });
+
+#if defined(CONDUIT_USE_OPENMP)
+    conduit::execution::forall(openmp, 0, size, [=] (int i)
+    {
+       std::cout << i << std::endl;
+    });
+#endif
+
+#if defined(CONDUIT_USE_RAJA)
+    conduit::execution::forall(device, 0, size, [=] (int i)
+    {
+       std::cout << i << std::endl;
+    });
+#endif
+
+    std::cout << "functor cases!" << std::endl;
+
+    MyFunctor func;
+    func.size = size;
+    conduit::execution::dispatch(serial, func);
+    std::cout << func.res << std::endl;
+
+#if defined(CONDUIT_USE_OPENMP)
```

Comment:
instead of ifdefs for these cases, can we ask the exec env if these are supported.
That tests we have the right runtime vs compile time interface

Reply:
done